File: structy-008.js
// Write a function, anagrams, that takes in two strings as arguments. The function 
// should return a boolean indicating whether or not the strings are anagrams. 
// Anagrams are strings that contain the same characters, but in any order.

const anagrams = (s1, s2) => {

let count = {};

for (char of s1) {
  if (!(char in count)) {
    count[char] = 0
  }
  count[char] += 1;
}

for (char of s2) {
  if (char in count) {
    count[char] -= 1;
  }
}
  let sum = null;

  for (let char in count) {
    if (count[char] !== 0) {
      return false;
    }
  }
  
  return true;
};

console.log(anagrams('monkeyswrite', 'newyorktimes'));
console.log(anagrams('cats', 'tocs'));
console.log(anagrams('paper', 'reapa')); 

File: structy-008.py
# Write a function, anagrams, that takes in two strings as arguments. The function 
# should return a boolean indicating whether or not the strings are anagrams. 
# Anagrams are strings that contain the same characters, but in any order.

# Using dictionaries:

# def anagrams(s1, s2):
#   return char_count(s1) == char_count(s2)

# def char_count(s):
#   count = {}
  
#   for char in s:
#     if char not in count:
#       count[char] = 0
#     count[char] += 1
  
#   return count

# Using sorted:

def anagrams(s1, s2):
    return sorted(s1) == sorted(s2)

print (anagrams('restful', 'fluster'))
